pkg/api/defaults.go

```diff
@@ -434,6 +434,10 @@ func (cs *ContainerService) setOrchestratorDefaults(isUpgrade, isScale bool) {
 			o.KubernetesConfig.ContainerRuntimeConfig = make(map[string]string)
 		}
 
+		if o.KubernetesConfig.KubeReservedCgroup == "" {
```

Comment:
Based on this, it is actually not possible for a fully populated (post-defaults enforcement) api model to have a `""` value of `KubeReservedCgroup`. In which case, the `HasKubeReservedCgroup` func is redundant, right?

Reply:
whoops, I'll remove this

Reply:
This is leftover from playing around with per-nodepool and defaulting, but it didn't work out the way i'd like

Reply:
That explains everything! :)